tangelo/linq/circuit.py

```diff
@@ -652,24 +653,100 @@ def __init__(self):
         """instantiate classical control operations"""
 
     @abc.abstractmethod
-    def return_circuit(self, measurement) -> List[Gate]:
-        """Return a circuit based on the measurement outcome.
+    def return_gates(self, measurement) -> List[Gate]:
+        """Return the list of gates based on the measurement outcome.
 
         Args:
             measurement (str): "1" or "0"
             qubit (int): The qubit index
 
         Returns:
-            Circuit: The next circuit to apply
+            List[Gate]: The next gates to apply to the circuit
         """
 
     @abc.abstractmethod
-    def finalize(self, frequencies):
+    def finalize(self):
         """Called from simulator after all gates have been called.
 
         Can be used to reinitialize variables for the next run
-        and store results
-
-        Args:
-            frequencies (dict): The measured frequencies for the final state measurement.
+        and store results.
         """
+
+
+def generate_applied_gates(source_circuit: Circuit, desired_meas_result=None) -> List[Gate]:
+    """Generate the applied gates of a Circuit without explicitly simulating.
+
+    Useful to determine the resources required for a circuit with measurement controlled operations given certain
+    measurement outcomes.
+
+    Note: Measurement outcomes with zero probability can not be screened.
+
+    Args:
+        source_circuit (Circuit): a circuit in the abstract format to be simulated with
+        no state but the Classical Control function called.
```

Comment:
Maybe my English is lacking but the alignment and the previous sentence make this flow a bit oddly. Are you sure 685-686 are complete sentences ? Multilines should be indented.